packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageContentStyles.ts

```diff
@@ -0,0 +1,56 @@
+import { ComponentSlotStylesPrepared, ICSSInJSStyle } from '@fluentui/styles';
+import { defaultChatDensity } from '../../../../components/Chat/chatDensity';
+
+import { ChatMessageContentStylesProps } from '../../../../components/Chat/ChatMessageContent';
+import { ChatMessageVariables } from './chatMessageVariables';
+import { pxToRem } from '../../../../utils';
+
+const getChatMessageVariantStyles = (props: ChatMessageContentStylesProps) => {
+  const density = props.density || defaultChatDensity;
+  switch (true) {
+    case props.layout === 'refresh' && density === 'comfy':
+      return ({ props: p, variables: v, theme }): ICSSInJSStyle => ({
+        color: v.contentColor,
+        wordBreak: 'break-word',
+        wordWrap: 'break-word',
+        '& a': {
+          color: 'inherit',
+          textDecoration: 'underline',
+          wordBreak: 'break-all',
+          '&:hover': { textDecorationStyle: 'double' },
+          '&:focus': { textDecorationStyle: 'double' },
+        },
+        ...(p.failed && {
+          color: theme.siteVariables.colorScheme.default.foreground,
+        }),
+      });
+    case density === 'comfy':
+      return ({ props: p }): ICSSInJSStyle => ({
+        ...(p.hasBadge && p.badgePosition === 'end' && { marginRight: pxToRem(4) }),
+      });
+
+    default:
+      return () => ({});
+  }
+};
```

Comment:
TBH I don't see a reason to a closure there anymore

```ts
const getChatMessageVariantStyles: ComponentSlotStyleFunction<ChatMessageContentStylesProps, ChatMessageVariables> = ({
  props: p,
  variables: v,
  theme,
}) => {
  return {
    ...(p.layout === 'refresh' &&
      p.density === 'comfy' && {
        color: v.contentColor,
        wordBreak: 'break-word',
        wordWrap: 'break-word',
        '& a': {
          color: 'inherit',
          textDecoration: 'underline',
          wordBreak: 'break-all',
          '&:hover': { textDecorationStyle: 'double' },
          '&:focus': { textDecorationStyle: 'double' },
        },
        ...(p.failed && {
          color: theme.siteVariables.colorScheme.default.foreground,
        }),
      }),

    ...(p.density === 'comfy' && {
      ...(p.hasBadge && p.badgePosition === 'end' && { marginRight: pxToRem(4) }),
    }),
  };
};
```

`const density = props.density || defaultChatDensity;` is also redundant as it should come from `defaultProps` then